### PR TITLE
json::File

### PR DESCRIPTION
**Pull request by t-h2o**
### Json::File

This class open a JSON file at the construction of an instance
and close it at the desctruction.

The `getc()` member function return a character.
getc()` do not return any blank character.

### Why ?

This class will split the JSON parser

### EDIT

I added:
* `get_string()`
* `get_number()`

### CODE REVIEW

**Review by RobinBurri — APPROVED** [review]
👍

File: src/json/File.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if (_file.is_open() == 0)
+		throw std::runtime_error("Error: read(): file is not valid");
+
+	getline(_file, _current_line);
+}
+
+File::~File(void) { _file.close(); }
+
+char
+File::getc(void)
```
Comment (0xCAF3D0OD) [inline src/json/File.cpp:18]: i don't understand the purpose of this function

Comment (RobinBurri) [reply src/json/File.cpp:18]: * The getc() member function return a character.
* getc() do not return any blank character.

Comment (0xCAF3D0OD) [reply src/json/File.cpp]: 👍

**Review by RobinBurri — APPROVED** [review]
ok

File: inc/File.hpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+namespace json
+{
+
+class File
+{
+  public:
+	File(const char *);
+	~File(void);
```
Comment (0xCAF3D0OD) [inline inc/File.hpp:22]: the class may not be canonical ?

Comment (t-h2o) [reply inc/File.hpp:22]: Why this class should be canonical ?

**Review by RobinBurri — APPROVED** [review]
OK

**Review by 0xCAF3D0OD — APPROVED** [review]
RAS

**Pull request merged**